submit.go

```diff
@@ -321,7 +329,8 @@ func parseJSONRequest(w http.ResponseWriter, req *http.Request, reportDir string
 	parsed.AppName = p.AppName
 
 	if p.UserAgent != "" {
-		parsed.Data["User-Agent"] = p.UserAgent
+		parsed.Data["User-Agent"] = parseUserAgent(p.UserAgent)
+		parsed.Data["Raw-User-Agent"] = p.UserAgent
```

Comment:
This feels slightly magical to me: I think I'd prefer to keep `User-Agent` as the raw string, and have (say) `Parsed-User-Agent` for the parsed version.